Put back deprecation warnings

waynew opened the pull request:
### What does this PR do?

Fixes #62185 - now we emit DeprecationWarnings again.

Can test against the MCVE in https://github.com/saltstack/salt/pull/59917

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- ~[x] Docs~
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- ~[ ] Tests written/updated~

### Commits signed with GPG?
Yes

waynew committed:
Put back deprecation warnings

Checked the
https://docs.python.org/3/library/warnings.html#warnings.filterwarnings
docs but there wasn't any obvious reason why our warnings disappear.
This restores the DeprecationWarnings showing up on the command line.

waynew committed:
Add tests for deprecation warnings

Rather than just rely on the test suite, we also add a deprecation
warning to the test module, which will enable a simple method for
ensuring that DeprecationWarnings are correctly emitted.

Commit:
Merge branch 'master' into fix/warnings/restore-deprecation

Commit:
Merge branch 'master' into fix/warnings/restore-deprecation

Commit:
Merge branch 'master' into fix/warnings/restore-deprecation

s0undt3ch reviewed (changes requested):
Test failures look valid.

garethgreenaway committed:
Merge branch 'master' into fix/warnings/restore-deprecation

waynew committed:
Include certain deprecation warnings on tests

Due to some of the versions of dependencies we're using, these tests may
emit some specific errors.

waynew committed:
Merge branch 'master' into fix/warnings/restore-deprecation

waynew commented:
@s0undt3ch test failures look to be sorted, finally :upside_down_face:
